astrowidgets/bqplot.py

```diff
@@ -0,0 +1,655 @@
+import numpy as np
+
+from astropy.coordinates import SkyCoord
+from astropy.io import fits
+from astropy.nddata import CCDData
+from astropy.table import Table, vstack
+from astropy import units as u
+import astropy.visualization as apviz
+
+from bqplot import Figure, LinearScale, Axis, ColorScale, PanZoom, ScatterGL
+from bqplot_image_gl import ImageGL
+from bqplot_image_gl.interacts import (MouseInteraction,
+                                       keyboard_events, mouse_events)
+
+import ipywidgets as ipw
+
+from matplotlib import cm as cmp
+from matplotlib import pyplot
+from matplotlib.colors import to_hex
+
+import traitlets as trait
+
+# Allowed locations for cursor display
+ALLOWED_CURSOR_LOCATIONS = ['top', 'bottom', None]
+
+# List of marker names that are for internal use only
+RESERVED_MARKER_SET_NAMES = ['all']
+
+
+class _AstroImage(ipw.VBox):
```

Comment:
It is really confusing to have the API split between `_AstroImage` and the official `ImageWidget`.

Reply:
I was trying to put as little as possible in `ImageWidget` that depended on the details of the backend implementation in the hope that we could factor some backend-indpenedent stuff out. The dividing line between the two got a little unclear, though.

Reply:
Yeah... At some point in my past effort, I just ended up not splitting them but rather hide the backend specific stuff as with leading underscore. 😹